elastic/kibana
# [Snapshot & Restore] - Show error messages in snapshots tab

Fixes: https://github.com/elastic/kibana/issues/98782

## Summary

Changes the warning callout to include all errors returned by the API, where as before it was just showing a hardcoded single case.

Note for reviewers: Will be easier to review the PR if you hide whitespaces.

**How to test**
1. Start elasticsearch with `yarn es snapshot --license=trial -E path.repo=/tmp/es-backups` and kibana with `yarn start`
2. Navigate to `Stack Management` -> `Snaphot restore` -> `Repository`, and create a `shared file system` repo with the location set to `/tmp/-es-backups`
3. Mock an error in the snapshots API by downloading the following patch [mock_error.diff.txt](https://github.com/elastic/kibana/files/10794271/mock_error.diff.txt) and applying it from the root of your kibana project with `git apply mock_error.diff.txt`
5. Verify that:
    * When there are no snapshots to be shown but there are errors it should show an empty prompt + callout with errors list
    * When there are snapshots and errors, the snapshots list and errors callout should be shown together

<details>
<summary>Screenshots</summary>

// With snapshot errors and snapshots to be shown on list
![Screenshot 2023-02-21 at 14 40 33](https://user-images.githubusercontent.com/1191206/220368988-b38d2012-e954-45cf-8ad0-e72f07f20382.png)

// With snapshot errors but no remaining snapshots to show on list
![Screenshot 2023-02-27 at 13 32 37](https://user-images.githubusercontent.com/1191206/221564783-a437222f-cc9b-4f05-9de0-33b2df089be4.png)


</details>
